src/boat_simulator/boat_simulator/nodes/physics_engine/physics_engine_node.py

```diff
@@ -120,6 +121,12 @@ def __init_private_attributes(self):
             generator=MVGaussianGenerator(np.array([1, 1]), np.array([[2, 1], [1, 2]]))
         )
 
+    def __update_wind_sensor(self):
+        """Updates the wind sensor with the latest wind data from the wind generator."""
+        wind_data = self.__wind_generator.next()
+        self.__wind_sensor.speed.speed = wind_data[0]
```

Comment:
I just happened to stumble upon this. You shouldn't have to assign the private members in this way. The update method is there to help you out, and you get additional functionality with noise/delays if you want. Try `self.update(wind=wind_data[0])`?

See the [wind sensor tests](https://github.com/UBCSailbot/sailbot_workspace/blob/058ebe4e9769b6e82230b492e7720e2e067e642a/src/boat_simulator/tests/unit/common/test_wind_sensor.py#L47) for an example of this type of call.